internal/stores/hostdb.go

```diff
@@ -206,6 +272,13 @@ func (db *SQLStore) ProcessConsensusChange(cc modules.ConsensusChange) {
 	}
 }
 
+func ExcludeBlockedHosts(db *gorm.DB) *gorm.DB {
+	return db.
```

Comment:
this is awful, should probably use something like `NOT EXISTS` here - checking that out